webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/S3DataVault.scala

```diff
@@ -143,12 +146,22 @@ object S3DataVault {
         new AnonymousAWSCredentialsProvider
     }
 
-  private def getAmazonS3Client(credentialOpt: Option[S3AccessKeyCredential]): AmazonS3 =
-    AmazonS3ClientBuilder.standard
+  private def getAmazonS3Client(credentialOpt: Option[S3AccessKeyCredential], uri: URI): AmazonS3 = {
+    val basic = AmazonS3ClientBuilder.standard
       .withCredentials(getCredentialsProvider(credentialOpt))
-      .withRegion(Regions.DEFAULT_REGION)
       .withForceGlobalBucketAccessEnabled(true)
-      .build
+    if (isPathStyle(uri) && !uri.getHost.endsWith(".amazonaws.com"))
```

Comment:
Could you extract this if condition into a isNonAmazonEndpoint or something? I think this current variant is fairly hard to read